src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -47,6 +47,8 @@ public partial class BlockTree : IBlockTree
         private readonly IDb _metadataDb;
         private readonly IBlockStore _badBlockStore;
 
+        private readonly PotentialCensorshipCache _potentialCensorshipCache;
```

Comment:
I would prefer if this lived as a separate component, outside of BlockTree. It can react to `BlockAddedToMain` event.